### PR TITLE
Add advanced features ana

### PR DESCRIPTION
**Pull request by CapatinaAna**
Added the 8 new operations

### CODE REVIEW

File: Calculator/Form1.cs [hunk 1]
```diff
@@ -295,6 +295,133 @@ private void PerformOperation()
...
+                        break;
+                    }
+                case Operation.IsPrime:
+                    {
+                        uint number = Convert.ToUInt32(firstTerm);
+                        if (number == 0 || number == 1)
+                        {
+                            lbl_title.Text = $"{number} is not a prime number.";
+                        }
+                        for (uint divisor = 2; divisor <= Math.Sqrt(firstTerm); divisor++)
```
Comment (AvramRotar) [inline Calculator/Form1.cs]: This for should be executed only if number is different from 0 or 1, please wrap it in an else statement

File: Calculator/Form1.cs [hunk 2]
```diff
@@ -295,6 +295,133 @@ private void PerformOperation()
...
+                        {
+                            if (number % divisor == 0)
```
Comment (AvramRotar) [inline Calculator/Form1.cs]: this is actually a bug. by this algorithm, 18 will appear to be a prime number, which is false.
Please think about what's wrong and fix

File: Calculator/Form1.cs [hunk 3]
```diff
@@ -295,6 +295,133 @@ private void PerformOperation()
...
+                            else
+                            {
+                                lbl_title.Text = $"{number} is  a prime number.";
+                            }
+                        }
+                        break;
+                    }
+                case Operation.PalindromeSuperPalindrome:
+                    {
+                        uint number = Convert.ToUInt32(firstTerm);
```
Comment (AvramRotar) [inline Calculator/Form1.cs]: There is no point to keep converting to uint here if you're not doing any validation

**Review by AvramRotar — CHANGES_REQUESTED** [review]

File: Calculator/Form1.cs (hunk 3 above)
Comment (CapatinaAna) [reply Calculator/Form1.cs]: Went to mirrored method and converted the uint result to double then deleted the conversions i had before in the PalindromeSuperPalindrome method

Comment (CapatinaAna) [reply Calculator/Form1.cs]: to GetMirrored*

File: Calculator/Form1.cs (hunk 1 above)
Comment (CapatinaAna) [reply Calculator/Form1.cs]: added else statement before the for statement, was it to easy so its wrong

Comment (CapatinaAna) [reply Calculator/Form1.cs]: ?*

File: Calculator/Form1.cs (hunk 2 above)
Comment (CapatinaAna) [reply Calculator/Form1.cs]: did some debbugging and realised i shouldve put a break  before the else, it works now

**Pull request closed without merging**